tensorflow/lite/tools/benchmark/external_delegate_provider.cc

```diff
@@ -0,0 +1,158 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/lite/tools/benchmark/benchmark_model.h"
+#include "tensorflow/lite/tools/benchmark/delegate_provider.h"
+#include "tensorflow/lite/tools/benchmark/logging.h"
+
+#include <dlfcn.h>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace tflite {
+namespace benchmark {
+namespace {
+// Split a given string to a vector of string using a delimiter character
+std::vector<std::string> SplitString(const std::string& str, char delimiter) {
+  std::vector<std::string> tokens;
+  std::string token;
+  std::istringstream ss(str);
+  while (std::getline(ss, token, delimiter)) {
+    tokens.push_back(token);
+  }
+  return tokens;
+}
+
+// External delegate library construct
+struct ExternalLib {
+  using CreateDelegatePtr = std::add_pointer<TfLiteDelegate*(
+      const char**, const char**, size_t,
+      void (*report_error)(const char*))>::type;
+  using DestroyDelegatePtr = std::add_pointer<void(TfLiteDelegate*)>::type;
+
+  // Open a given delegate library and load the create/destroy symbols
+  void load(const std::string library) {
+    if (!is_loaded) {
+      void* handle = nullptr;
+      handle = dlopen(library.c_str(), RTLD_LAZY | RTLD_LOCAL);
+      if (handle == nullptr) {
+        TFLITE_LOG(INFO) << "Unable to load external delegate from : "
+                         << library;
+      } else {
+        create = reinterpret_cast<decltype(create)>(
+            dlsym(handle, "tflite_plugin_create_delegate"));
```

Comment:
We do support non-posix platforms, so we'll either need to restrict this in such cases or use a generic intermediate which toggles between LoadLibrary/GetProcAccess

Reply:
Can I use `tensorflow/tensorflow/core/platform/load_library.h`  ?

Reply:
The TF platform targets aren't generally mobile portable. We can just use a lightweight shim here, e.g., https://github.com/tensorflow/tensorflow/blob/4911ca4a011c7b86087e4a420247fdc675ad8a88/tensorflow/lite/python/interpreter_wrapper/interpreter_wrapper.cc#L158. We can refactor later into a more portable helper.